### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Anwesh1**
Using history server to grab PMID's instead of I/O process.

-> Updated ```query_key``` to get values from ESEARCH
-> Getting data in batches of 100,000 instead of original 10,000

### CODE REVIEW

**Comment by Anwesh1** [comment]
hey @jvwong, do you want me to use ```str.splitlines()``` for the lines on the history server? 
here to be specific - https://github.com/PathwayCommons/pubmed-dl/blob/c97c57fd78de18747ce86dfae60a6af7d0a1a4a4/ncbi.py#L141-L142

**Comment by Anwesh1** [comment]
The ```str.splitlines()``` is actually cool; makes it much easier. I will push with that update.

File: ncbi.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    fetch_url = f"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?&retmax=100000&query_key={query}&db=pubmed&rettype=uilist&retmode=text&WebEnv={webenv}"
+    print(f"Total records: {str(total)}")
+    count = 1
+    store = []
+    for i in range(0, total, 100000):
+        this_fetch = f"{fetch_url}&retstart={str(i)}"
+        print(f"URL {str(count)}: {this_fetch}")
+        count += 1
+        fetch_r = requests.post(this_fetch)
+        for word in fetch_r.text.split():
```
Comment (jvwong) [inline ncbi.py]: Do you think this would work: `store += fetch_r.text.splitlines()` ?

File: ncbi.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    fetch_url = f"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?&retmax=100000&query_key={query}&db=pubmed&rettype=uilist&retmode=text&WebEnv={webenv}"
+    print(f"Total records: {str(total)}")
+    count = 1
+    store = []
+    for i in range(0, total, 100000):
+        this_fetch = f"{fetch_url}&retstart={str(i)}"
+        print(f"URL {str(count)}: {this_fetch}")
+        count += 1
+        fetch_r = requests.post(this_fetch)
+        store = str.splitlines(fetch_r.text)
```
Comment (jvwong) [inline ncbi.py]: Did you mean `store +=` ?

**Review by jvwong — CHANGES_REQUESTED** [review]

File: pubmed_dl/ncbi.py [hunk 3]
```diff
@@ -123,44 +123,20 @@ def uids_to_docs(uids: List[str]) -> List[Dict[str, Collection[Any]]]:
...
-            if word == "pmid":
-                p_flag = True
-            elif p_flag:
-                p_flag = False
-                if word[:-1] not in store:
-                    store.append(word[:-1])
-        f.close()
-    for i in range(0, total, 10000):
-        os.remove("pubmed_batch_" + str(i) + "_to_" + str(i + 9999) + ".json")
+        store = str.splitlines(fetch_r.text)
```
Comment (jvwong) [inline pubmed_dl/ncbi.py]: I don't think this is correct - `fetch_r.text.splitlines()`

Comment (Anwesh1) [reply pubmed_dl/ncbi.py]: It is working the way I'm doing it right now ```str.splitlines(fetch_r.text)```, but thanks for the '+=' catch. Do you want me to change it to ```fetch_r.text.splitlines()``` ?

Comment (Anwesh1) [reply pubmed_dl/ncbi.py]: it works both ways apparently. I tried them.

Comment (jvwong) [reply pubmed_dl/ncbi.py]: Ok I guess `str` is a global var. Cool.

**Comment by jvwong** [comment]
Do I need to do something special to run at command line? I'm getting this error:

```
> pubmed-dl
...

ModuleNotFoundError: No module named 'ncbi'
```

**Comment by JohnGiorgi** [comment]
@jvwong You would have had to run `pip install -e .` (see [here](https://github.com/PathwayCommons/pubmed-dl#installing-the-library-and-dependencies)). If you did that and still get the error let me know and I will look into it!

**Comment by jvwong** [comment]
I still get that error . I'm on `PMID-retrieval-updated` branch. 
I can run it fine on `main` branch.

File: pubmed_dl/main.py [hunk 4]
```diff
@@ -1,20 +1,20 @@
-#!/usr/bin/env python
 import json
-
-from pubmed_dl.ncbi import get_list_pmid, uids_to_docs
```
Comment (JohnGiorgi) [inline pubmed_dl/main.py:4]: This shouldn't have been deleted. This is what's causing @jvwong 's error.

We have formatted this as a python package, so you should specify the absolute path in your import statements. See [here](https://realpython.com/python-import/) if you are unfamiliar with pythons importing and how it works (it's a little wacky).

We should add some tests with `pytest` in another PR so this kind of stuff gets caught.

**Review by JohnGiorgi — CHANGES_REQUESTED** [review]

File: pubmed_dl/main.py (hunk 4 above)
Comment (jvwong) [reply pubmed_dl/main.py:4]: Yes that fixed it thanks!

Comment (Anwesh1) [reply pubmed_dl/main.py:4]: Oh oops, I have matched the imports to the ```main.py``` file on the ```main``` branch. Should work now I think.

Comment (Anwesh1) [reply pubmed_dl/main.py:4]: However, this change is causing some merge checks to be unsuccessful. Is there something wrong??

**Comment by Anwesh1** [comment]
Is everything good? Let me know and I can merge.

**Review by jvwong — APPROVED** [review]
Looks good to me

**Pull request merged**